feat(editor): Add 'Whats new' section and modal

Cadiac opened the pull request:
## Summary

This PR introduces a new "What's new" section to the front page of n8n. This section displays articles showcasing new features and other news we want to showcase to our users. These articles are loaded from creators portal's backend and their markdown is rendered.

https://github.com/user-attachments/assets/80d544d8-6aab-4c0b-a0e5-3e39fd02e88d

Read "What's new" articles are tracked in local storage, and the notification icon is shown if there are either version updates available or user hasn't seen all articles.

"Update" button either takes the user to the instructions to update their self-hosted instance or on the cloud dashboard, depending on their hosting platform.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/ADO-3685/feature-whats-new-in-n8n-frontend

<!--
Include links to **Linear ticket** or Github issue or Community forum post.
Important in order to close *automatically* and provide context to reviewers.
https://linear.app/n8n/issue/
-->
<!-- Use "closes #<issue-number>", "fixes #<issue-number>", or "resolves #<issue-number>" to automatically close issues when the PR is merged. -->


## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

Cadiac committed:
feat(editor): Fetch what's new articles on page load

Cadiac committed:
feat(editor): Add What's new section to the left hand side menu

Cadiac committed:
feat(editor): Fix some styles issues on menuItems

Cadiac committed:
refactor(editor): Simplify the menuitem styles, remove unread property

Cadiac committed:
feat(editor): Make N8nMenu support custom menu items

Cadiac committed:
feat(editor): Style the update CTA button better

Cadiac committed:
feat(editor): Better check for displaying update CTA

Cadiac committed:
feat(editor): Change the whats-new endpoint implementation, handle it being down gracefully

Cadiac committed:
feat(editor): Allow '100%' or other string values to be passed to youtube embed iframes

Cadiac committed:
feat(editor): Style the WhatsNewModal a bit

Cadiac committed:
feat(editor): Show a version updates callout if new versions are available

Cadiac committed:
feat(editor): Small improvements

Cadiac committed:
feat(editor): Store read 'What's new articles' on local storage

Cadiac committed:
feat(editor): Disable the update button if up to date

Cadiac committed:
fix(editor): Fix scroller missing clicked articles

Cadiac committed:
feat(editor): Show light ellipses next to read articles, simplify menuitem icon coloring

Cadiac committed:
feat(editor): Anchor the update callout to the top of the what's new content

Cadiac committed:
fix(editor): Make paddings on the popper submenu more consistent

Cadiac committed:
feat(editor): Add env to toggle what's new feature, menu stories

Cadiac committed:
test(editor): Add tests for WhatsNewModal component

Cadiac committed:
test(editor): Add tests for VersionUpdateCTA component

Cadiac committed:
test(editor): Add tests for versions store

Cadiac commented on packages/frontend/@n8n/design-system/src/components/N8nMenuItem/MenuItem.vue:
```diff
@@ -261,12 +310,53 @@ const N8nMenuItem = getCurrentInstance()?.type;
 	}
+
+	&.small {
+		font-size: var(--font-size-2xs) !important;
```
Yes, these `!important`s on this file hurt me too :(

MiloradFilipovic replied:
Yeah but looks like it's needed with all of the other `!important`s already there :/ We need to clean this up

Cadiac committed:
test(editor): Fix config test

Cadiac committed:
fix(editor): Fix type errors

Cadiac committed:
test(editor): Fix E2E versions tests

Cadiac committed:
test(editor): Address some issues at tests

MiloradFilipovic commented on packages/frontend/editor-ui/src/components/MainSidebar.vue:
```diff
@@ -280,6 +319,20 @@ const handleSelect = (key: string) => {
 			telemetry.track('User clicked insights link from side menu');
 		default:
+			if (key.startsWith('whats-new-article-')) {
+				const articleId = Number(key.replace('whats-new-article-', ''));
```
Are `NaN` values expected here? Should we add a bit od error handling?

Cadiac replied:
I can't really see any way this would error - we only create these at the mapping function below, and all articles have a required integer ID on them. As long as we don't introduce other elements with the same prefix we're fine.

MiloradFilipovic commented on packages/frontend/editor-ui/src/components/VersionUpdateCTA.vue:
```diff
@@ -0,0 +1,75 @@
+	margin-bottom: var(--spacing-3xs);
+	border-radius: var(--border-radius-base);
+	border: var(--border-base);
+	border-color: var(--color-foreground-base);
```
Nit: Isn't color already set to `-color-foreground-base` in `--border-base`?

MiloradFilipovic commented on packages/frontend/editor-ui/src/components/WhatsNewModal.vue:
```diff
@@ -0,0 +1,264 @@
+	await pageRedirectionHelper.goToVersions();
+};
+
+watch(
```
When are these updated? Do we really need to watch for changes or can we just pass them to the component?

Cadiac replied:
With the current implementation they shouldn't really change. I was after a "on component mount, mark the articles visible on this modal read" implementation, how would you rather see that done?

MiloradFilipovic commented on packages/frontend/editor-ui/src/components/WhatsNewModal.vue:
```diff
@@ -0,0 +1,264 @@
+		:show-close="false"
+	>
+		<template #header>
+			<div :class="$style.header">
```
Spacing and alignment look a bit off to me here. Would maybe look better if bell icon is top-aligned
![image](https://github.com/user-attachments/assets/b5ec610d-7595-4546-9c98-d267c18c81de)

Cadiac replied:
I was hoping to get some designs for this but didn't, and this was left at my placeholder implementation :D Nik had some comments on this too so I'll discuss with him tomorrow on what we wnat this to look like.

MiloradFilipovic reviewed (dismissed):
Looks very good to me 🚀 Left a few comments, mostly nitpicks.

Cadiac committed:
feat(editor): Disallow html by default on markdown, allow it on What's new

We've previously allowed HTML on markdown but manually escaped it, since the very first
commit where this functionality was introduced. Markdown-it suggests using the html: false
mode if html rendering isn't needed, and it does the &gt escaping for us, so our own custom
escapeMarkdown function is unnecessary. (Dangerously) enable HTML rendering on the What's new
modal, the contents are controlled by us and anyways passed through 'xss' so simple script tags
etc still won't lead to XSS, but we can support custom html elements like collapsible lists.

Cadiac committed:
refactor(editor): Remove unnecessary style

Cadiac committed:
refactor(editor): Use onMounted to mark what's new items read

MiloradFilipovic commented on packages/frontend/editor-ui/src/components/WhatsNewModal.vue:
```diff
@@ -47,25 +47,21 @@ const onUpdateClick = async () => {
 	scroller.value?.scrollToItem(articleIndex);
 });
+
+onMounted(() => {
```
This looks better, but this logic doesn't really has to do anything with the components, it just updates the state in the store based on the state in the store. So I would suggest moving it to the versions store and setting read articles after they are fetched in the `checkForNewVersions` method. We can also create `init` method in versions store that does these things (potentially other init logic) and is called from the `init.ts`

MiloradFilipovic reviewed (commented):
Great cleanup, I left a comment on the new `onMounted` hook so take a look when you can.

Cadiac committed:
feat(editor): Add comment on marking what's new items read

MiloradFilipovic reviewed (dismissed):
🚀

Cadiac committed:
Merge branch 'master' into ado-3685-feature-whats-new-in-n8n-frontend

Cadiac committed:
fix(editor): Don't show the update callout if user is up to date

Cadiac committed:
Merge branch 'master' into ado-3685-feature-whats-new-in-n8n-frontend

janober commented:
Got released with [`n8n@1.101.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.101.0)
